### ISSUE TITLE
Compilation error : Unexpected expression intern.string_intern(...)

### ISSUE THREAD

**Issue by Argc0** [ryzhyk/differential-datalog#186]
I have this small program: 
```
import intern
typedef Symbol = IString

relation BMethod(b1:Symbol, b2:Symbol)
relation CMethod(c1:Symbol, c2:Symbol)
relation AMethod(a1:Symbol, a2:Symbol)

CMethod(a, b) :- BMethod(string_intern("c"), b), AMethod(b, a).
```
while I am trying to compile it I get an error:
```
ddlog: Unexpected expression intern.string_intern("c") in Compile.arrangeInput.subst'
CallStack (from HasCallStack):
  error, called at src/Language/DifferentialDatalog/Compile.hs:1612:32 in differential-datalog-0.1.0.0-1kbn6LIs0fJLftbNrKD4ud:Language.DifferentialDatalog.Compile
```
If I alternate the relations of the body there is no compilation error and if I use the string type instead of IString also works.
I try to understand, why this error occured?

**Comment by ryzhyk** [comment]
This is a bug. Thanks for reporting! Will try to fix later today.

**Comment by ryzhyk** [comment]
@Argc0, the bug should be gone now, let us know if this unblocks you.

**Comment by Argc0** [comment]
Well, when I am trying to compile the above program via ddlog compilation (```ddlog -i <program>.dl --action=compile -L lib```) the error still remains...and I can't explain why.

**Comment by ryzhyk** [comment]
Did you pull the latest master and run `stack install`?

**Comment by ryzhyk** [comment]
In fact your example is not part of the test suite, so should definitely work:

https://github.com/ryzhyk/differential-datalog/blob/d25283527817db5f09b0b1636aed9513a6cd0aba/test/datalog_tests/simple.dl#L818

**Comment by mbudiu-vmw** [comment]
He meant "is now part of the testsuite"

**Comment by mbudiu-vmw** [comment]
I suspect he is using the released code.

**Comment by ryzhyk** [comment]
If that's the case, maybe it's time to roll out v0.4 with all the recent improvements

**Comment by Argc0** [comment]
I am so sorry, I was running "stack build" and it seems that did not change the ddlog executable.
